### PR TITLE
storage: add metrics for KV and DO request timings

### PR DESCRIPTION
**Pull request by Noah-Kennedy**

### CODE REVIEW

**Review by lbaquerofierro — APPROVED** [review]

**Review by cjpatton — CHANGES_REQUESTED** [review]
Please fix the clippy warnings before merging.

**Comment by Noah-Kennedy** [comment]
> Please fix the clippy warnings before merging.

One step ahead of you!

**Review by cjpatton — APPROVED** [review]

File: crates/daphne-worker/src/storage_proxy/mod.rs [hunk 1]
```diff
@@ -210,13 +210,33 @@ fn parse_expiration_header(ctx: &RequestContext) -> Result<Option<Time>, worker:
 
 /// Handle a kv request.
 async fn handle_kv_request(ctx: &mut RequestContext<'_>, key: &str) -> worker::Result<Response> {
+    let start = std::time::Instant::now();
```
Comment (RageKnify) [inline crates/daphne-worker/src/storage_proxy/mod.rs:213]: Shouldn't this be `Worker::Date::now()`? I believe WASM doesn't implement the `std::time` module. For plexi I used `Worker::Date::now()` to measure the time to process requests.

Comment (mendess) [reply crates/daphne-worker/src/storage_proxy/mod.rs:213]: I believe it should be. Some docs on the subject https://developers.cloudflare.com/workers/runtime-apis/performance/

Comment (mendess) [reply crates/daphne-worker/src/storage_proxy/mod.rs:213]: You were right, check #647 on how to forbid that error from ever compiling

**Pull request merged**